### PR TITLE
Feature/453827 organisation registration decision handling

### PR DESCRIPTION
**Pull request by JulianYoungEviden**
### Support Regulator Organisation Registration Decision Submission Events
This change allows creating a new event in the SubmissionEvents container.
A new EventType has been added, RegulatorOrganisationRegistratioDecision, with value 12.  This is because it cannot overload the existing RegulatorDecision as that is for submissions with Files.
Save and Get operations exist, with the Get command support a SubmissionID filter.
The Swagger UI has been specified to present a meaningful test datum.

### CODE REVIEW

**Review by johnfitzgeraldEviden — DISMISSED** [review]

**Review by shehzad-ismail8 — DISMISSED** [review]

File: src/EPR.SubmissionMicroservice.Application.UnitTests/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionCreateCommandValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        command.SubmissionId = Guid.NewGuid();
+
+        _mockQueryRepository
+            .Setup(x => x.GetByIdAsync(command.SubmissionId, It.IsAny<CancellationToken>())).ReturnsAsync((Submission)null);
+
+        // Act
+        var result = await _systemUnderTest.TestValidateAsync(command);
+        result.Should().NotBeNull();
+        // Assert
+        //result.ShouldHaveValidationErrorFor(x => x.SubmissionId)
```
Comment (shehzad-ismail8) [inline src/EPR.SubmissionMicroservice.Application.UnitTests/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionCreateCommandValidatorTests.cs]: Are these comments required?

File: src/EPR.SubmissionMicroservice.Application/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using EPR.SubmissionMicroservice.Data.Repositories.Queries.Interfaces;
+using FluentValidation;
+
+namespace EPR.SubmissionMicroservice.Application.Features.Commands.SubmissionEventCreate.Validators;
+
+public class RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator : AbstractValidator<RegulatorOrganisationRegistrationDecisionEventCreateCommand>
+{
+    public RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator(IQueryRepository<Submission> queryRepository)
+    {
+        //Include(new SubmissionEventCreateCommandValidator(queryRepository));
```
Comment (shehzad-ismail8) [inline src/EPR.SubmissionMicroservice.Application/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator.cs]: Is this comment required?

**Review by shehzad-ismail8 — CHANGES_REQUESTED** [review]
A couple of comments

File: src/EPR.SubmissionMicroservice.Application.UnitTests/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionCreateCommandValidatorTests.cs (hunk 1 above)
Comment (JulianYoungEviden) [reply src/EPR.SubmissionMicroservice.Application.UnitTests/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionCreateCommandValidatorTests.cs]: The code will be reverted I expect

File: src/EPR.SubmissionMicroservice.Application/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator.cs (hunk 2 above)
Comment (JulianYoungEviden) [reply src/EPR.SubmissionMicroservice.Application/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator.cs]: The code will be reverted I expect

File: src/EPR.SubmissionMicroservice.Application.UnitTests/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionCreateCommandValidatorTests.cs (hunk 1 above)
Comment (shehzad-ismail8) [reply src/EPR.SubmissionMicroservice.Application.UnitTests/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionCreateCommandValidatorTests.cs]: Okay

File: src/EPR.SubmissionMicroservice.Application/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator.cs (hunk 2 above)
Comment (shehzad-ismail8) [reply src/EPR.SubmissionMicroservice.Application/Features/Commands/SubmissionEventCreate/Validators/RegulatorOrganisationRegistrationDecisionEventCreateCommandValidator.cs]: Okay

**Review by shehzad-ismail8 — DISMISSED** [review]

**Review by johnfitzgeraldEviden — DISMISSED** [review]

**Review by shehzad-ismail8 — APPROVED** [review]

**Review by johnfitzgeraldEviden — APPROVED** [review]

**Review by MasilEviden — APPROVED** [review]

**Pull request merged**